src/Savegame/BattleUnit.cpp

```diff
@@ -5053,6 +5053,15 @@ void getFactionScript(const BattleUnit *bu, int &faction)
 	faction = 0;
 }
 
+void setSpawnUnitScript(BattleUnit *bu, ScriptText unitType)
+{
+	if (bu)
+	{
+		bu->setSpawnUnit(unitType);
```

Comment:
This have one problem, this could crash game if you use incorrect type and I would prefer if scripts can't crash game, even if script is incorrect.

Probably better solution would be first load unit to spawn from mod class and then set it to unit.
This could be even opportunity for small refactor, instead of storing string, store unit rule of spawned unit in affected unit. This would allow querying it in scripts and check some conditions on it.

Reply:
Another question, what if we want remove spawn unit? Simply reset it to empty. This could be possible functionality to add.

Reply:
How does setSpawnUnitScript get access to a Mod Pointer?